### PR TITLE
feat: add shuffle method

### PR DESCRIPTION
**Pull request by Demivan**
Adds `shuffle` method. Creates an array of shuffled values, using the [Fisher-Yates shuffle](https://en.wikipedia.org/wiki/Fisher-Yates_shuffle).

---

Make sure that you:

- [x] Typedoc added for new methods and updated for changed
- [x] Tests added for new methods and updated for changed
- [x] New methods added to `src/index.ts`
- [x] New methods added to `mapping.md`

### CODE REVIEW

File: src/shuffle.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * [Fisher-Yates shuffle](https://en.wikipedia.org/wiki/Fisher-Yates_shuffle).
+ * @param items the array to shuffle
+ * @signature
+ *    R.shuffle(array)
+ * @example
+ *    R.shuffle(['one', 'two', 'three']) // => ['two', 'three', 'one']
+ * @category Array
+ */
+export function shuffle<T>(items: ReadonlyArray<T>): Array<T> {
+  const length = items.length
```
Comment (eranhirsch) [inline src/shuffle.ts:12]: I don't think we need to memoize the length prop of the array, I can't imagine this having an impact on perf over accessing items.length directly

**Review by eranhirsch — CHANGES_REQUESTED** [review]
I feel this is over-verbose for what it needs to be.

**Comment by eranhirsch** [comment]
Also I just noticed this doesn't provide a data-last version

```typescript
function shuffle<T>(): (data: ReadonlyArray<T>) => Array<T>;
```

**Comment by Demivan** [comment]
@eranhirsch Thanks for review. I have implemented your comments.

**Comment by eranhirsch** [comment]
The code has non-consistent semi-colons, can you run `yarn lint && yarn prettier` on the branch?

**Comment by Demivan** [comment]
Done

**Review by eranhirsch — APPROVED** [review]

File: src/shuffle.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { purry } from './purry';
+
+/**
+ * Creates an array of shuffled values, using the
+ * [Fisher-Yates shuffle](https://en.wikipedia.org/wiki/Fisher-Yates_shuffle).
```
Comment (eranhirsch) [inline src/shuffle.ts:5]: No need to document the algorithm used, that is an internal that isn't relevant to the caller, and would become stale if we ever implement a different shuffling algorithm.

File: src/shuffle.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { purry } from './purry';
+
+/**
+ * Creates an array of shuffled values, using the
```
Comment (eranhirsch) [inline src/shuffle.ts:4]: I'd phrase this as "Shuffles the input array" as this doesn't create an array per-se (like you'd expect a "randomArray" function would create one)

**Review by eranhirsch — APPROVED** [review]
nits about the docs

File: src/shuffle.ts (hunk 3 above)
Comment (Demivan) [reply src/shuffle.ts:4]: I wanted to indicate that it does not mutate the original array. "Shuffles the input array" sounds like it is mutating.

Comment (eranhirsch) [reply src/shuffle.ts:4]: All remeda functions shouldn't mutate the input, that's a core concept of Remeda.

Comment (Demivan) [reply src/shuffle.ts]: Updated

**Pull request merged**